### PR TITLE
Refactor Progress component

### PR DESCRIPTION
**Pull request by leandrotk**
## What is this PR?

Refactoring the `Progress` component. The whole idea is to separate concerns. The component will only render the HTML and the functions will handle logic and data.

## PR changes

- Remove some code: static method and component default state
- Extract the progress style logic into a function & cleanup the component
- Add a ProgressWrapper component

## PS

It would be great to add [unit tests](https://github.com/mohitk05/react-insta-stories/issues/18) to these functions and components.

### CODE REVIEW

**Comment by mohitk05** [comment]
Hey @leandrotk, the function `getDerivedStateFromProps ` in Progress component is necessary to determine the duration of the next incoming story (mostly if it is a video). That cannot be removed. Please make the changes or let me know if there's any other explanation to this.

**Comment by leandrotk** [comment]
@mohitk05, add it back again.

Can you show me how it is used? I couldn't find any method calling it.

**Comment by mohitk05** [comment]
@leandrotk this function is a lifecycle method in React and is called before every render. It returns a state object to update the state.
https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

**Comment by leandrotk** [comment]
@mohitk05 Good to know! :)

I added it back now! :)

**Comment by leandrotk** [comment]
@mohitk05 Good to know! :)

I added it back now! Let me know if I need to change anything else.

File: src/components/Progress.js [hunk 1]
```diff
@@ -34,35 +28,28 @@ export default class Progress extends React.PureComponent {
...
-        innerStyle = { width: 0 }
-        break
-    }
+    const { width, pause, bufferAction, active, defaultInterval } = this.props
     return (
-      <div className={style.autoHide} style={{...styles.progress, ...{width: `${this.props.width * 100}%`, opacity: this.props.pause && !this.props.bufferAction ? 0 : 1}}}>
-        <div ref={r => { this.inner = r }} className={style.inner} style={innerStyle} /* style={{...styles.overlay, width: `${this.props.completed * 100}%`}} */ />
-      </div>
+      <ProgressWrapper width={width} pause={pause} bufferAction={bufferAction}>
+        <div ref={r => { this.inner = r }} className={style.inner} style={getProgressStyle({ active, defaultInterval, pause })} />
```
Comment (ovionlogis) [inline src/components/Progress.js]: instead of the `defaultInterval` you need to pass `this.state.duration`

Comment (ovionlogis) [reply src/components/Progress.js]: also, in the functions below (`animation`, `getProgressStyle`), I'd rather rename the argument `defaultInterval` to `duration`, it makes more sense

Comment (leandrotk) [reply src/components/Progress.js]: @ovionlogis Thanks for the feedback. I've just updated it.

**Comment by leandrotk** [comment]
@ovionlogis @mohitk05 Updated the code!

**Comment by mohitk05** [comment]
Thanks for the PR @leandrotk ! Sorry for the delayed merge. Keep contributing 💫

**Pull request merged**